aten/src/THC/THCTensorRandom.cu

```diff
@@ -143,7 +154,7 @@ GENERATE_KERNEL2(generate_cauchy, float, double median, double sigma, float, cur
 GENERATE_KERNEL2(generate_cauchy, double, double median, double sigma, double, curand_uniform_double, (double)(median + sigma * tan(M_PI*(x-0.5))))
 
 #ifdef CUDA_HALF_TENSOR
-GENERATE_KERNEL2(generate_uniform, half, double a, double b, float, curand_uniform, (ScalarConvert<float, half>::to(x * (b-a) + a)))
+GENERATE_KERNEL2(generate_uniform, half, double a, double b, float, curand_uniform, (__float2half_rd(reverse_bounds(x) * (b-a) + a)))
```

Comment:
Looks like this breaks for CUDA 7.5 (`__float2half_rd` is undefined).
Are we still supporting CUDA 7.5?

Reply:
yea we need to fix it.